mlflow/server/js/src/experiment-tracking/pages/experiment-overview/components/TraceLatencyChart.test.tsx

```diff
@@ -0,0 +1,375 @@
+import { jest, describe, it, expect, beforeEach } from '@jest/globals';
+import { screen, waitFor } from '@testing-library/react';
+import { renderWithIntl } from '../../../../common/utils/TestUtils.react18';
+import { TraceLatencyChart } from './TraceLatencyChart';
+import { DesignSystemProvider } from '@databricks/design-system';
+import { QueryClient, QueryClientProvider } from '@mlflow/mlflow/src/common/utils/reactQueryHooks';
+import {
+  MetricViewType,
+  AggregationType,
+  TraceMetricKey,
+  P50,
+  P90,
+  P99,
+  getPercentileKey,
+} from '@databricks/web-shared/model-trace-explorer';
+
+// Mock FetchUtils
+jest.mock('../../../../common/utils/FetchUtils', () => ({
+  fetchOrFail: jest.fn(),
+  getAjaxUrl: (url: string) => url,
+}));
+
+import { fetchOrFail } from '../../../../common/utils/FetchUtils';
+const mockFetchOrFail = fetchOrFail as jest.MockedFunction<typeof fetchOrFail>;
+
+// Helper to create mock API response
+const mockApiResponse = (dataPoints: any[] | undefined) => {
+  mockFetchOrFail.mockResolvedValue({
+    json: () => Promise.resolve({ data_points: dataPoints }),
+  } as Response);
+};
+
+// Helper to chain mock API responses (for percentiles + AVG calls)
+const mockApiResponses = (percentileDataPoints: any[], avgDataPoints: any[]) => {
+  mockFetchOrFail
+    .mockResolvedValueOnce({
+      json: () => Promise.resolve({ data_points: percentileDataPoints }),
+    } as Response)
+    .mockResolvedValueOnce({
+      json: () => Promise.resolve({ data_points: avgDataPoints }),
+    } as Response);
+};
+
+// Helper to create a latency percentile data point
+const createLatencyDataPoint = (timeBucket: string, p50: number, p90: number, p99: number) => ({
+  metric_name: TraceMetricKey.LATENCY,
+  dimensions: { time_bucket: timeBucket },
+  values: {
+    [getPercentileKey(P50)]: p50,
+    [getPercentileKey(P90)]: p90,
+    [getPercentileKey(P99)]: p99,
+  },
+});
+
+// Helper to create an AVG latency data point
+const createAvgLatencyDataPoint = (avg: number) => ({
+  metric_name: TraceMetricKey.LATENCY,
+  dimensions: {},
+  values: { [AggregationType.AVG]: avg },
+});
+
+// Mock recharts components to avoid rendering issues in tests
+jest.mock('recharts', () => ({
```

Comment:
idea: maybe we could externalize it to be reused in other recharts-powered component tests